src/sagemaker_tensorflow_container/training.py

```diff
@@ -196,9 +208,11 @@ def main():
     # If the training job is part of the multiple training jobs for tuning, we need to append the training job name to
     # model_dir in case they read from/write to the same object
     if '_tuning_objective_metric' in hyperparameters:
-        env.hyperparameters['model_dir'] = os.path.join(hyperparameters.get('model_dir'), env.job_name, 'checkpoints')
+        model_dir = _model_dir_with_training_job(hyperparameters.get('model_dir'), env.job_name)
+        logger.info('Appending the training job name to model_dir: {}'.format(model_dir))
+    else:
+        model_dir = hyperparameters.get('model_dir')
```

Comment:
It feels a little awkward for the normal training job case. We are getting model_dir from hyperparameters then later on sets hyperparameters['model_dir'] to the same value. I would just make model_dir optional in _run_worker. YMMV.

Reply:
we would then have to call `framework.env.read_hyperparameters()` again (see my note in the description) - not sure if it's particularly expensive to do that?

Reply:
how about we just pass the cmd_args in _run_worker instead of model_dir?